libs/libcoap/Makefile

```diff
@@ -66,11 +63,11 @@ TARGET_CFLAGS += $(FPIC)
 
 CONFIGURE_ARGS += \
 	--enable-examples \
-	--disable-documentation
-
-ifeq ($(CONFIG_BIG_ENDIAN),y)
-TARGET_CFLAGS += -DWORDS_BIGENDIAN
-endif
+	--disable-documentation \
+	--disable-doxygen \
+	--disable-dtls \
+	--disable-gcov \
+	--disable-tests
```

Comment:
As I can see from the version 4.2.0, **tests** are disabled by default configuration. The same is **gcov**. Please, correct me if it's not.

Reply:
Correct. I only made it explicit.

Reply:
I haven't looked at OpenWRT's packages conventions for a while, @karlp could you please clarify if it make sense to add some flags explicitly?

Reply:
I have no strong opinion on that :)  I would personally not bother until it caused problems, but that's just me.  Given that we _know_ we never want documents and tests (which often try and "run" things which never work for cross builds) it's probably a pretty safe bet to explicitly disable tests and gcov.  I wouldn't say you should explicitly set _every_ default though, and I that's not what neheb has done, so this is fine with me at least, but  you're the maintainer of this package :)

Reply:
One benefit is that if upstream changes their mind, we would still be disabling them.